pkg/client/client.go

```diff
@@ -229,21 +366,76 @@ func (c *SPDKClient) EngineWatch(ctx context.Context) (*api.EngineStream, error)
 	return api.NewEngineStream(stream), nil
 }
 
-func (c *SPDKClient) EngineReplicaDelete(engineName, replicaName, replicaAddress string) error {
+func (c *SPDKClient) EngineSnapshotCreate(name, snapshotName string) error {
+	if name == "" || snapshotName == "" {
+		return fmt.Errorf("failed to create SPDK engine snapshot: missing required parameter name or snapshot name")
+	}
+
 	client := c.getSPDKServiceClient()
 	ctx, cancel := context.WithTimeout(context.Background(), GRPCServiceTimeout)
 	defer cancel()
 
-	_, err := client.EngineReplicaDelete(ctx, &spdkrpc.EngineReplicaDeleteRequest{
+	_, err := client.EngineSnapshotCreate(ctx, &spdkrpc.SnapshotRequest{
+		Name:         name,
+		SnapshotName: snapshotName,
+	})
+	return errors.Wrapf(err, "failed to create SPDK engine %s snapshot %s", name, snapshotName)
+}
+
+func (c *SPDKClient) EngineSnapshotDelete(name, snapshotName string) error {
+	if name == "" || snapshotName == "" {
+		return fmt.Errorf("failed to delete SPDK engine snapshot: missing required parameter name or snapshot name")
+	}
+
+	client := c.getSPDKServiceClient()
+	ctx, cancel := context.WithTimeout(context.Background(), GRPCServiceTimeout)
+	defer cancel()
+
+	_, err := client.EngineSnapshotDelete(ctx, &spdkrpc.SnapshotRequest{
+		Name:         name,
+		SnapshotName: snapshotName,
+	})
+	return errors.Wrapf(err, "failed to delete SPDK engine %s snapshot %s", name, snapshotName)
+}
+
+func (c *SPDKClient) EngineReplicaAdd(engineName, replicaName, replicaAddress string) error {
```

Comment:
It's better to make time-consuming operations async. This can be improved later. Basically, have a call operation, then stats operation to understand the status.

Reply:
The legacy rebuilding API is a sync call as well. It's better to make the change simultaneously.

Reply:
I see. Let's improve it later.